Tasks/Kubernetes/task.json

```diff
@@ -0,0 +1,207 @@
+            "visibleRule": "containerregistrytype = Container Registry",
+            "groupName": "registry"
+        },
+        {
+            "name": "azureSubscriptionEndpoint",
+            "type": "connectedService:AzureRM",
+            "label": "Azure subscription",
+            "helpMarkDown": "Select an Azure subscription",
+            "visibleRule": "containerregistrytype = Azure Container Registry",
+            "required": true,
+            "groupName": "registry"
+        },
+        {
+            "name": "azureContainerRegistry",
+            "label": "Azure Container Registry",
+            "type": "pickList",
+            "helpMarkDown": "Select an Azure Container Registry",
+            "visibleRule": "containerregistrytype = Azure Container Registry",
+            "required": true,
+            "defaultValue": "",
+            "groupName": "registry"
+        },
+        {
+            "name": "secretName",
+            "type": "string",
+            "label": "Secret name",
+            "required": false,
+            "defaultValue": "",
+            "helpMarkDown": "Secret name",
+            "groupName": "registry"
+        },
+        {
+            "name": "forceUpdate",
+            "type": "boolean",
+            "label": "Force Update secret",
+            "defaultValue": "true",
+            "helpMarkDown": "Recreate secret.",
+            "groupName": "registry"
+        },
+        {
+            "name": "command",
+            "type": "pickList",
+            "label": "Command",
+            "defaultValue": "apply",
+            "required": true,
+            "options": {
+                "apply": "apply",
+                "create": "create",
+                "delete": "delete",
+                "exec": "exec",
+                "expose": "expose",
+                "get": "get",
+                "logs": "logs",
+                "run": "run",
+                "set": "set",
+                "top": "top"
+            },
+            "helpMarkDown": "Select a kubectl command.",
+            "groupName": "commands",
+            "properties": {
+                "EditableOptions": "True"
+            }
+        },
+        {
+            "name": "useConfigurationFile",
+            "type": "boolean",
+            "label": "Use Configuration files",
+            "defaultValue": "false",
+            "helpMarkDown": "Use configuration file for the command.",
+            "groupName": "commands"
+        },
+        {
+            "name": "configuration",
+            "type": "filePath",
+            "label": "Configuration File",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Path to the Configuration file or directory.",
+            "visibleRule": "useConfigurationFile = true",
+            "groupName": "commands"
+        },
+        {
+            "name": "arguments",
```

Comment:
verifed that tool runner is handling special character like ' ,'@' etc aant the command execution fails with no such command found error.